docs/code-guide.md

```diff
@@ -0,0 +1,133 @@
+# Coding Guidelines
+
+This document defines the coding standards for the OCM UI project. All code contributions must follow these guidelines to ensure consistency, maintainability, and code quality.
+
+These guidelines serve to:
+- Establish shared conventions for code structure and patterns
+- Improve code reviews by setting clear expectations
+- Maintain a consistent, maintainable codebase
+
+Use these guidelines during development and code reviews. When in doubt or if a scenario isn't covered, ask the team for help.
+
+## Component Structure
+
+Components should be organized by features. Each feature is represented by a container component in charge of handling logic and rendering other presentational components. The main responsibilities of presentational components are to display data received via props and handle user interactions.
+
+Components should be properly broken down into UI elements responsible for a single goal. 
+
+Example folder structure for a hypothetical `UsersList` component:
+
+```
+UsersList/
+├── UsersList.tsx           # Container component
+├── UsersList.test.tsx      
+├── UsersList.stories.tsx      
+├── useUsersList.ts         # Feature-specific hooks
+├── components/             # Presentational components (only imported by container)
+├── └── UsersTable.tsx      # Only create a folder if you have many
+├── └── UsersToolbar.tsx
+├── types.ts                # Only if needed      
+└── utils/                  # Feature-specific utilities
+```
+
+## Coding Patterns
```

Comment:
Do you think we should attach examples for the recommended coding patterns from our codebase to inspect? It would help the team understand the coding pattern better, when it is connected to our project (if we have a good example for it of course 😇)

Reply:
If you mean linking code in our repo, that's brittle. Files will change, move, we'll have to maintain those links.

We could write examples from scratch here eventually. Which coding patterns do you think require better explanation? I would add them only where it's necessary to avoid making the document too long to read (it's already too long, I know!)